### ISSUE TITLE
User isn't made aware when a form elements appear and disappear on conditional logic

### ISSUE THREAD

**Issue by adriannelee** [cds-snc/design-gc-conception#2090]
Canada Holidays
https://fable-test.d2fzvkr34ak02t.amplifyapp.com/
2025-12-05
Fail
**Description of Issue (required):**

Screen readers do not inform users of the results of certain actions.

Example: On the "Submit a holiday" page, clicking the "Other" option under "What type of holiday is this?" will result in a question appearing but screen reader users will not be informed of this. Likewise click the other option again, will result in it disappear and likewise screen reader users will not be informed.

**Impact on Users:**

Users of assistive technologies will not be made aware of new or changed content if screen readers do not inform them of it.

**Sample Code** (required):

Other button: <label for="other" class="gcds-label">…</label>

The new question that appears:

<label for="province" class="gcds-label">…</label>
**Recommendation (required if not SAMS/OCMC):**

When a status message advises on the success or results of an action, or the state of an application, we recommend using role=status to present status messages.

**Additional Resources:**

https://www.w3.org/WAI/WCAG21/Understanding/status-messages.html https://www.w3.org/WAI/WCAG21/Techniques/aria/ARIA22 https://www.w3.org/WAI/WCAG21/Techniques/general/G199
**Steps to Reproduce:**

1. Open the solution on the "Submit a holiday" page: https://fable-test.d2fzvkr34ak02t.amplifyapp.com/submit-a-holiday/

2. Turn on a screen reader such as NVDA.

3. Click the "Other" option under "What type of holiday is this?"

4. This creates a new question below of "If this holiday occurs in a specific province or territory, select the location.(required)".

5. Click the "Other" option again and the question will disappear.

Actual result:

Screen reader users are not informed of this change.

Expected result:

Screen reader users are informed of this change.
